solutions/usaco-1018.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: usaco-1018
+source: USACO Gold 2020 February
+title: Help Yourself
+author: Aakash Gokhale
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_help_gold_feb20.html)
+
+## Solution
+
+The brute force approach of this problem would be to iterate over all subsets and count the number of connected components for each subset.
+
+However, this approach has a time complexity of $\mathcal{O}(2^N * N)$ and is too large to fit the time constraints of the problem except for the first three test cases. To optimize this, we can figure out how much each segment contributes to the total answer.
+
+Consider the sample test case
+
+![](usaco-1018/help_yourself_sample_illustration.png)
+
+For each segment we can list the subsets where the segment is counted. We have to be careful about overcounting so we will only list the subsets where the segment is the beginning of a connected componenet.
```

Comment:
```suggestion
For each segment we can list the subsets where the segment is counted. We must only list the subsets where the segment is the beginning of a connected component, since we would overcount otherwise.